modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -559,15 +559,10 @@ def build_query_compiler(cls, dataset, columns, index_columns, **kwargs):
         )
         index, sync_index = cls.build_index(dataset, partition_ids, index_columns)
         remote_parts = cls.build_partition(partition_ids, column_widths)
-        if len(partition_ids) > 0:
-            row_lengths = [part.length() for part in remote_parts.T[0]]
-        else:
-            row_lengths = None
```

Comment:
I don't see how there could be any performance hit, as we initialize the partitions with their respective sizes here, so `.length()` should be just returning immediately:
https://github.com/modin-project/modin/blob/7871c7bc5adeeaca9b6fc4b88482be8b18333faa/modin/core/io/column_stores/parquet_dispatcher.py#L466-L478

Reply:
I was thinking the same thing too. There was some discussion before about this, but we tabled the discussion for another time. `length` should just be returning the object reference here right? I don't think anything will be getting materialized.